File: aforclinic-api-main/aforclinic-api-main/src/main/java/com/xenialsoft/api/core/beforeafter/model/BeforeAfter.java
package com.xenialsoft.api.core.beforeafter.model;

import java.util.List;

import com.xenialsoft.api.common.Auditable;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class BeforeAfter extends Auditable {

	private String id;
	
	private String title;
	
	private String description;
	
	private String image;
	
	private String beforeImage;
	
	private String afterImage;
	
	private String afterBlurImage;
	
	private BeforeAfterStatus status;
	
	private List<BeforeAfterCategories> beforeAfterCategoriesList;
	
	private List<BeforeAfterTags> beforeAfterTagsList;	
}

File: aforclinic-web-main/aforclinic-web-main/shared/types/BeforeAfter.ts
export type BeforeAfterBannerMediaType =
  'IMAGE' |
  'VIDEO';

export interface BeforeAfterBanner {
  /** 전후사진 배너 식별자 */
  id: string;
  /** 전후사진 배너 미디어 타입 */
  mediaType: BeforeAfterBannerMediaType;
  /** 전후사진 배너 미디어 정보 */
  mediaContent: string;
}

export interface BeforeAfter {
  /** 전후사진 식별자 */
  id: string;
  /** 전후사진 제목 */
  title: string;
  /** 전후사진 설명 */
  description: string;
  /** 전후사진 */
  image: string;
  /** 전 사진 */
  beforeImage?: string;
  /** 후 사진 */
  afterImage?: string;
}
